src/components/common/NetworkSelector/NetworkMultiSelector.tsx

```diff
@@ -36,7 +38,7 @@ const NetworkMultiSelector = ({
     (chains: ChainInfo[]) => {
       if (chains.length !== 1) return
       const shortName = chains[0].shortName
-      const networkLink = getNetworkLink(router, shortName, isWalletConnected)
+      const networkLink = getNetworkLink(router, safeAddress, shortName)
       router.replace(networkLink)
     },
     [isWalletConnected, router],
```

Comment:
The `safeAddress` is missing in the dependencies.

Reply:
Good catch!